pvsite_datamodel/write/user_and_site.py

```diff
@@ -245,6 +246,45 @@ def update_user_site_group(session: Session, email: str, site_group_name: str) -
     return user
 
 
+# update site metadata
+def edit_site(
+    session: Session, site_uuid: str, site_info: PVSiteEditMetadata
+) -> Tuple[SiteSQL, str]:
+    """
+    Edit an existing site. Fill in only the fields that need to be updated.
+
+    :param session: database session
+    :param site_uuid: the existing site uuid
+    :param site_info: metadata for editing the site, should be an instance of PVSiteEditMetadata
+        - client_site_id: id the client uses for the site
+        - client_site_name: name the client uses for the site
+        - orientation: orientation of site
+        - tilt: tilt of site
+        - latitude: latitude of site as an integer
+        - longitude: longitude of site as an integer
+        - inverter_capacity_kw: inverter capacity of site in kw
+        - module_capacity_kw: module capacity of site in kw
+        - capacity_kw: capacity of site in kw
+        - dno: dno of site
+        - gsp: gsp of site
+    """
+    site = session.query(SiteSQL).filter(SiteSQL.site_uuid == site_uuid).first()
+
+    update_data = site_info.model_dump(exclude_unset=True, exclude_none=False)
```

Comment:
nice, so does, drop all the unset values, but keep them if they have been set to None?

Reply:
Exactly.